platform_plugin_aspects/views.py

```diff
@@ -58,9 +67,29 @@ def has_object_permission(self, request, view, obj):
 Course = namedtuple("Course", ["course_id", "display_name"])
 
 
-class SupersetView(GenericAPIView):
+def _get_course(course_key):
     """
-    Superset-related endpoints provided by the aspects platform plugin.
+    Return a Course-like object for the requested course id.
+
+    Raise NotFound if the corresponding course does not exist.
+    """
+    # Fetch the CourseOverview (if we're running in edx-platform)
+    display_name = ""
+    CourseOverview = get_model("course_overviews")
+    if CourseOverview:
+        try:
+            course_overview = CourseOverview.objects.get(id=course_key)
```

Comment:
`CourseOverview` is a pretty big model with a lot of fields, if you only care about display_name you could do (fake code, but should be close):

`course_overview = CourseOverview.objects.get(id=course_key).only("display_name")` to get the model with only that field populated or

`display_name = CourseOverview.objects.get(id=course_key).values("display_name")["display_name"]` to skip all of the model code and just get return the name.

This matters more now with likely increase in how often this code will be called.